### PR TITLE
fix(1-1): fix contact label not updating when mutual contact changes

### LINKED ISSUE #5709: Contact details are not being reflected without restart
### Description

1. add a desktop account to contacts from mobile device and check incoming contact requests -> nothing is shown without restart
2. add a desktop account to contacts from mobile device and confirm on desktop (so mobile should be your contact now) -> it is shown as `not in contacts` until restart

image

### PR DESCRIPTION
**Pull request by jrainville**
Fixes #5709

### CODE REVIEW

**Comment by jrainville** [comment]
@igor-sirotin you're the lucky person that has to test

**Review by saledjenic — APPROVED** [review]

File: src/app_service/service/contacts/dto/contacts.nim [hunk 1]
```diff
@@ -120,7 +120,7 @@ proc isBlocked*(self: ContactsDto): bool =
 proc isMutualContact*(self: ContactsDto): bool =
   # TODO not implemented in `status-go` yet
   # But for now we consider that contact is mutual contact if I added him and he added me.
-  return self.hasAddedUs and self.added
+  return self.hasAddedUs and self.added and not self.removed and not self.blocked
```
[the remark this answers is not in the document]
Comment (jrainville) [reply src/app_service/service/contacts/dto/contacts.nim:123]: Good find. I updated it.

**Comment by jrainville** [comment]
@igor-sirotin I just tried again and it worked for me. Can you give it another shot please?

If it still doesn't work, can you give me the steps you used?

**Review by elina2015 — APPROVED** [review]
Tested in https://ethstatus.testrail.net/index.php?/runs/view/9247&group_by=tests:status_id&group_order=asc

**Comment by elina2015** [comment]
@igor-sirotin thank you for the detailed steps for reproducing the issue. The issue with the deleted and returned contact and the label "Not a contact' is stable and reproduced now and it should be fixed. So, I'm reopening this task. 
image
image

@jrainville we need a fix from you:)

**Comment by jrainville** [comment]
Just an FYI: I rebased this PR and tried to reproduce the error described above, but it seems like Contact requests are pretty broken and being refactored/improved at the moment as part of https://github.com/status-im/status-desktop/issues/6114

I'll wait until accepting contact requests is fixed in master before fixing this further

**Comment by jrainville** [comment]
I rebased again. I was able to reproduce the issue you listed @igor-sirotin . I'm fixing it right now

**Comment by jrainville** [comment]
Cool, gonna mark as tested since it's a bug fix and you tested yourself.

**Pull request merged**